### PR TITLE
Added example to demonstrate using HardwareSerial with RS485 interfaces

### PR DESCRIPTION
**Pull request by econeale**
## Description of Change
This PR adds an example to the set of Serial examples that demonstrates how to use the HardwareSerial interface to communicate using RS485 hardware. The demo simply echos data received on one interface to the other. 

## Tests scenarios
I have tested with Adafruit Huzzah32 (ESP32) and Adafruit QT Py ESP32-S3 using a MAX33046E translation chip. 

## Related links
Addendum to #8934 
Potentially related to #8755

### CODE REVIEW

File: libraries/ESP32/examples/Serial/RS485_Echo_Demo/RS485_Echo_Demo.ino [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#define RS485_RTS_PIN 37
+
+HardwareSerial RS485(2);
+
+void setup() {
+  Serial.begin(9600);
+  delay(2000);
+
+  RS485.begin(9600, SERIAL_8N1, RS485_RX_PIN, RS485_TX_PIN);
+  if(!RS485.setPins(-1, -1, -1, RS485_RTS_PIN)){
```
Comment (lucasssvaz) [inline libraries/ESP32/examples/Serial/RS485_Echo_Demo/RS485_Echo_Demo.ino:29]:
```suggestion
  RS485.begin(9600, SERIAL_8N1, RS485_RX_PIN, RS485_TX_PIN);
  while(!RS485) { delay(10); }
  if(!RS485.setPins(-1, -1, -1, RS485_RTS_PIN)){
```

**Review by lucasssvaz — CHANGES_REQUESTED** [review]

**Review by lucasssvaz — APPROVED** [review]

File: libraries/ESP32/examples/Serial/RS485_Echo_Demo/RS485_Echo_Demo.ino [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  UART to RS485 translation hardware (e.g., MAX485, MAX33046E, ADM483) is assumed to be configured in half-duplex
+  mode with collision detection as described in
+  https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/uart.html#circuit-a-collision-detection-circuit
+
+  To use the script open the Arduino serial monitor (or alternative serial monitor on the Arduino port). Then,
+  using an RS485 tranciver, connect another serial monitor to the RS485 port. Entering data on one terminal
+  should be displayed on the other terminal.
+*/
+
+#include <HardwareSerial.h>
```
Comment (lucasssvaz) [inline libraries/ESP32/examples/Serial/RS485_Echo_Demo/RS485_Echo_Demo.ino:15]:
```suggestion
```

File: libraries/ESP32/examples/Serial/RS485_Echo_Demo/RS485_Echo_Demo.ino [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+#include "hal/uart_types.h"
+
+#define RS485_RX_PIN 16
+#define RS485_TX_PIN 5
+#define RS485_RTS_PIN 37
```
Comment (atanisoft) [inline libraries/ESP32/examples/Serial/RS485_Echo_Demo/RS485_Echo_Demo.ino:20]: pin 37 doesn't exist on all ESP32 variants, I'd suggest moving this to GPIO 4 or other IO pin that is generally available.

**Comment by VojtechBartoska** [comment]
Hi @econeale, can you please update your PR up to the reviews? Thanks a lot!

**Review by lucasssvaz — APPROVED** [review]

**Comment by SuGlider** [comment]
@me-no-dev - this PR can also be merged.

**Pull request merged**